node/rpc-client/src/lib.rs

```diff
@@ -1,12 +1,180 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-mod auth_ops;
-mod chain_ops;
-mod client;
-mod wallet_ops;
-
-pub use self::auth_ops::*;
-pub use self::chain_ops::*;
-pub use self::client::*;
-pub use self::wallet_ops::*;
+/// Filecoin RPC client interface methods
+pub mod auth_ops;
+pub mod chain_ops;
+pub mod net_ops;
+pub mod wallet_ops;
+
+/// Filecoin HTTP JSON-RPC client methods
+use jsonrpc_v2::{Error, Id, RequestObject, V2};
+use log::{debug, error};
+use parity_multiaddr::{Multiaddr, Protocol};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use std::env;
+
+const DEFAULT_MULTIADDRESS: &str = "/ip4/127.0.0.1/tcp/1234/http";
+const DEFAULT_URL: &str = "http://127.0.0.1:1234/rpc/v0";
+const DEFAULT_PROTOCOL: &str = "http";
+const DEFAULT_HOST: &str = "127.0.0.1";
+const DEFAULT_PORT: &str = "1234";
+const API_INFO_KEY: &str = "FULLNODE_API_INFO";
+const RPC_ENDPOINT: &str = "rpc/v0";
+
+/// Error object in a response
+#[derive(Deserialize)]
+pub struct JsonRpcError {
+    pub code: i64,
+    pub message: String,
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+pub enum JsonRpcResponse<R> {
+    Result {
+        jsonrpc: V2,
+        result: R,
+        id: Id,
+    },
+    Error {
+        jsonrpc: V2,
+        error: JsonRpcError,
+        id: Id,
+    },
+}
+
+struct URL {
+    protocol: String,
+    port: String,
+    host: String,
+}
+
+/// Parses a multiaddress into a URL
+fn multiaddress_to_url(ma_str: String) -> String {
+    // Parse Multiaddress string
+    let ma: Multiaddr = ma_str.parse().expect("Parse multiaddress");
```

Comment:
Why not return an error here?

Reply:
Actually, this isn't new code, this is moved code. Also, I'm completely refactoring how Multiaddrs are passed and parsed in the CLI in this PR:
https://github.com/ChainSafe/forest/pull/1183/files